datastore/google/cloud/datastore/client.py

```diff
@@ -157,26 +157,29 @@ class Client(_BaseClient, _ClientProjectMixin):
     :type namespace: str
     :param namespace: (optional) namespace to pass to proxied API methods.
 
-    :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
-                       :class:`NoneType`
-    :param credentials: The OAuth2 Credentials to use for the connection
-                        owned by this client. If not passed (and if no ``http``
-                        object is passed), falls back to the default inferred
-                        from the environment.
-
-    :type http: :class:`httplib2.Http` or class that defines ``request()``.
-    :param http: An optional HTTP object to make requests. If not passed, an
+    :type credentials: :class:`~google.auth.credentials.Credentials`
+    :param credentials: (Optional) The OAuth2 Credentials to use for this
+                        client. If not passed (and if no ``http`` object is
+                        passed), falls back to the default inferred from the
+                        environment.
+
+    :type http: :class:`~httplib2.Http`
+    :param http: (Optional) HTTP object to make requests. Can be any object
+                 that defines ``request()`` with the same interface as
+                 :meth:`~httplib2.Http.request`. If not passed, an
                  ``http`` object is created that is bound to the
                  ``credentials`` for the current object.
     """
-    _connection_class = Connection
 
     def __init__(self, project=None, namespace=None,
                  credentials=None, http=None):
         _ClientProjectMixin.__init__(self, project=project)
```

Comment:
Not an issue with this PR, but a  mix-in whose ctor messes up `super` is a code smell.